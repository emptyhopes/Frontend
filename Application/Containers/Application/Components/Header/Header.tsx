import styled from "styled-components";

import { Link } from "react-router-dom";

import { Container } from "@/Application/Containers/Application/Components/Container/Container";
import { Navbar } from "@/Application/Containers/Application/Components/Header/Components/Navbar/Navbar";

const StyledHeader = styled.header`
  background-color: #fffadd;
`;

const StyledContent = styled.div`
  display: flex;
  height: 120px;
  align-items: center;
  justify-content: space-between;
  padding: 10px 100px;
`;

const StyledLogo = styled.div`
  width: 200px;
  height: 100px;

  & img {
    width: 200px;
    height: 100px;
  }
`;

const Header = () => {
  return (
    <>
      <StyledHeader>
        <Container>
          <StyledContent>
            <StyledLogo>
              <Link to={"/"}>
                <img src="https://i.imgur.com/ZSHGk74.jpeg" alt="" />
              </Link>
            </StyledLogo>
            <Navbar />
          </StyledContent>
        </Container>
      </StyledHeader>
    </>
  );
};

export { Header };
