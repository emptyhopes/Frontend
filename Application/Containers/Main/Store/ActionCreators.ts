import { createAsyncThunk } from "@reduxjs/toolkit";

import axios from "axios";

const FetchUsers = createAsyncThunk("user/FetchUsers", async (_, ThunkAPI) => {
  try {
    const response = await axios.get("https://jsonplaceholder.typicode.com/users");
    return response.data;
  } catch (error) {
    return ThunkAPI.rejectWithValue("Произошла ошибка.");
  }
});

export { FetchUsers };
