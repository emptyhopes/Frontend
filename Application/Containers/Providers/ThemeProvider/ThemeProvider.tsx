import { ThemeProvider as Provider } from "styled-components";

import { UseApplicationSelector } from "@/Application/Ship/Store/Hooks/UseApplicationSelector";

import { PropsInterface } from "@/Application/Containers/Providers/ThemeProvider/Types/PropsInterface";

import { Light } from "@/Application/Ship/Themes/Light";
import { Dark } from "@/Application/Ship/Themes/Dark";

const ThemeProvider: React.FunctionComponent<PropsInterface> = (props: PropsInterface): React.ReactElement => {
  const theme = UseApplicationSelector((state) => state.ThemeReducer.theme);

  const GetTheme = () => {
    if (theme === "Light") return Light;
    if (theme === "Dark") return Dark;
  };

  return (
    <>
      <Provider theme={GetTheme()} {...props}></Provider>
    </>
  );
};

export { ThemeProvider };
