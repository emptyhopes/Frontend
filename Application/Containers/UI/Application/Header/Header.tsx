import styled from "styled-components";

import { Link, useLocation } from "react-router-dom";

import Logo from "@/Application/Ship/Assets/Images/logo.png";

import { Container } from "@/Application/Containers/UI/Application/Container/Container";

const StyledHeader = styled.header`
  flex: 0 0 auto;
  background-color: var(--header-background-color);
`;

const StyledContent = styled.div`
  display: flex;
  height: 120px;
  align-items: center;
  justify-content: space-between;
  padding: 0;
`;

const StyledLogo = styled.div`
  & img {
    width: 200px;
    height: 105px;
  }
`;

const StyledNavbar = styled.div`
  display: flex;
  align-items: center;
`;

const StyledUl = styled.ul`
  display: flex;
  padding: 0;

  & li + li {
    margin: 0 0 0 40px;
  }
`;

const StyledLi = styled.li`
  & a {
    padding: 10px 20px;
    border-radius: 20px;
    background-color: #ffffff;
    font-weight: 700;
  }

  & a:hover {
    color: #;
  }

  &.active {
    & a {
      background-color: #c0c0c0;
    }
  }
`;

const StyledLink = styled(Link)`
  padding: 10px 20px;
  border-radius: 20px;
  background-color: #ffffff;
`;

const Header: React.FunctionComponent = (): React.ReactElement => {
  const location = useLocation();

  const GetElementLink = (url: string, children: React.ReactNode) => {
    if (location.pathname === url) {
      return <StyledLi className={"active"}>{children}</StyledLi>;
    }

    return <StyledLi>{children}</StyledLi>;
  };

  return (
    <>
      <StyledHeader>
        <Container>
          <StyledContent>
            <StyledLogo>
              <Link aria-label="redirect to home page" to={"/"}>
                <img src={Logo} alt="" />
              </Link>
            </StyledLogo>
            <StyledNavbar>
              <nav>
                <StyledUl>
                  {GetElementLink("/", <StyledLink to={"/"}>Home</StyledLink>)}
                  {GetElementLink("/users", <StyledLink to={"/users"}>Users</StyledLink>)}
                  {GetElementLink("/posts", <StyledLink to={"/posts"}>Posts</StyledLink>)}
                  {GetElementLink("/todos", <StyledLink to={"/todos"}>Todos</StyledLink>)}
                </StyledUl>
              </nav>
            </StyledNavbar>
          </StyledContent>
        </Container>
      </StyledHeader>
    </>
  );
};

export { Header };
