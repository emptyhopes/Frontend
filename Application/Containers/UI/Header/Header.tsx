import styled from "styled-components";

import { Link } from "react-router-dom";

import Logo from "@/Application/Ship/Assets/Images/Logo.png";

import { Container } from "@/Application/Containers/UI/Container/Container";
import { Navbar } from "@/Application/Containers/UI/Header/Components/Navbar/Navbar";

const StyledHeader = styled.header`
  background-color: #fffadd;
`;

const StyledContent = styled.div`
  display: flex;
  height: 120px;
  align-items: center;
  justify-content: space-between;
  padding: 10px 100px;
`;

const StyledLogo = styled.div`
  width: 200px;
  height: 100px;

  & img {
    width: 200px;
    height: 100px;
  }
`;

const Header = () => {
  return (
    <>
      <StyledHeader>
        <Container>
          <StyledContent>
            <StyledLogo>
              <Link to={"/"}>
                <img src={Logo} />
              </Link>
            </StyledLogo>
            <Navbar />
          </StyledContent>
        </Container>
      </StyledHeader>
    </>
  );
};

export { Header };
