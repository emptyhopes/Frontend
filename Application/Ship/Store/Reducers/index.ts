import { combineReducers } from "@reduxjs/toolkit";

import { UsersMiddleware } from "@/Application/Containers/Middlewares/UsersMiddlewares/UsersMiddleware";
import { AuthenticationMiddleware } from "@/Application/Containers/Middlewares/AuthenticationMiddlewares/AuthenticationMiddleware";

import { AuthenticationReducer } from "@/Application/Containers/Store/Authentication/Slices/Slice";

const reducers = combineReducers({
  [UsersMiddleware.reducerPath]: UsersMiddleware.reducer,
  [AuthenticationMiddleware.reducerPath]: AuthenticationMiddleware.reducer,

  AuthenticationReducer,
});

export type RootState = ReturnType<typeof reducers>;

export { reducers };
