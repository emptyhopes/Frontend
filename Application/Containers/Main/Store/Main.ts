import { createSlice, PayloadAction } from "@reduxjs/toolkit";

import { State } from "@/Application/Containers/Main/Store/State/State";
import { FetchUsers } from "@/Application/Containers/Main/Store/ActionCreators";

const Main = createSlice({
  name: "Main",
  initialState: State,
  reducers: {
    increment(state) {
      state.counts = state.counts + 1;
    },
    decrement(state) {
      state.counts = state.counts - 1;
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(FetchUsers.pending.type, (state) => {
        state.isLoading = true;
      })
      .addCase(FetchUsers.rejected.type, (state, action: PayloadAction<string>) => {
        state.isLoading = false;
        state.error = action.payload;
      })
      .addCase(FetchUsers.fulfilled.type, (state, action: PayloadAction<[]>) => {
        state.isLoading = false;
        state.error = "";
        state.users = action.payload;
      });
  },
});

const MainReducers = Main.reducer;
const MainActions = Main.actions;

export { MainReducers, MainActions };
