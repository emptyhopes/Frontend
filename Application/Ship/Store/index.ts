import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";

import { reducers } from "@/Application/Ship/Store/Reducers/index";

import { UsersMiddleware } from "@/Application/Containers/Middlewares/UsersMiddlewares/UsersMiddleware";
import { AuthenticationMiddleware } from "@/Application/Containers/Middlewares/AuthenticationMiddlewares/AuthenticationMiddleware";

export const store = configureStore({
  reducer: reducers,

  middleware: [...getDefaultMiddleware(), UsersMiddleware.middleware, AuthenticationMiddleware.middleware],
});

export type ApplicationDispatch = typeof store.dispatch;
