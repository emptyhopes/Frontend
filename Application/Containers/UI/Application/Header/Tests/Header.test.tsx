import { render } from "@testing-library/react";

import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import { Header } from "@/Application/Containers/UI/Application/Header/Header";

import { store } from "@/Application/Ship/Store/index";

import { ThemeProvider } from "@/Application/Containers/Providers/ThemeProvider/ThemeProvider";

test("Header", () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <ThemeProvider>
          <Header />
        </ThemeProvider>
      </BrowserRouter>
    </Provider>,
  );

  expect(container.firstChild).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});
