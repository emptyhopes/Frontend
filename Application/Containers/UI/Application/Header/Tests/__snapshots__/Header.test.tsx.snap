// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 1`] = `
.c1 {
  width: 100%;
  max-width: 1240px;
  height: 100%;
  padding: 0 10px;
  margin: 0 auto;
}

.c0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  background-color: var(--header-background-color);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 120px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0;
}

.c3 img {
  width: 200px;
  height: 105px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
}

.c5 li + li {
  margin: 0 0 0 40px;
}

.c6 a {
  padding: 10px 20px;
  border-radius: 20px;
  background-color: #ffffff;
  font-weight: 700;
}

.c6 a:hover {
  color: #;
}

.c6.active a {
  background-color: #c0c0c0;
}

.c7 {
  padding: 10px 20px;
  border-radius: 20px;
  background-color: #ffffff;
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <a
            aria-label="redirect to home page"
            href="/"
          >
            <img
              alt=""
            />
          </a>
        </div>
        <div
          class="c4"
        >
          <nav>
            <ul
              class="c5"
            >
              <li
                class="c6 active"
              >
                <a
                  class="c7"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                class="c6"
              >
                <a
                  class="c7"
                  href="/users"
                >
                  Users
                </a>
              </li>
              <li
                class="c6"
              >
                <a
                  class="c7"
                  href="/posts"
                >
                  Posts
                </a>
              </li>
              <li
                class="c6"
              >
                <a
                  class="c7"
                  href="/todos"
                >
                  Todos
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </header>
</div>
`;
